tests/secrets/test_local_filesystem.py

```diff
@@ -222,9 +222,9 @@ def test_missing_file(self, mock_exists):
                 {
                     "conn_a": "mysql://hosta",
                     "conn_b": ''.join(
```

Comment:
You changed the connection setup here.
Before:
```
>> parse_qsl(urlparse(before).query)
[('extra__google_cloud_platform__keyfile_dict', "{'a': 'b'}"), ('extra__google_cloud_platform__keyfile_path', 'asaa')]
```
After:
```
>> parse_qsl(urlparse(after).query)
'{"extra__google_cloud_platform__keyfile_dict": {"a": "b"}, "extra__google_cloud_platform__keyfile_path": "asaa"}'
```
It is expected to pass JSON as text as the user enters the JSON as text. See: http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow-providers-google/latest/connections/gcp.html#configuring-the-connection

Reply:
Not quite.

I did not change the _connection_ at all.  Your yaml still parses to the same connection object.  But when parsed back to a URI, it actually became a different connection object; if you took that original uri and made it a connection, it would not have equalled the connection produced by your yaml, because it would have produced a different `extra`.

After my change, this would no longer be the case, since it can now represent the connection produced by your yaml without loss.

In my [most recent commit](https://github.com/apache/airflow/pull/15100/commits/7c6a02ae670ef5d87fedeb90e2f32598dbba0c5c#diff-ec0b1f1eed6ee36d376bf7340a10f5b7c6b57602ab56faf369b102482e9bd922L222-L229) I sidestep this issue entirely by comparing attributes instead of using get URI.  Now we test directly what your yaml conn parses to, without touching get_uri.

Reply:
The extra__google_cloud_platform__keyfile_dict hey should contain JSON as a string, so I think we can generate URL in old-way. We have only string as a extra fields value.

Reply:
> The extra__google_cloud_platform__keyfile_dict hey should contain JSON as a string
That is true, but your yaml example does not do this. The problem is the way you defined the yaml.  You need to put it as a string and not as an object if you want local filesystem backend to parse it as such.

Reply:
> so I think we can generate URL in old-way. 

if you change your yaml so that `extra__google_cloud_platform__keyfile_dict` is a string and not an object, then it's true the generated URI would be old-style.  but since it's not like that it was forced to use new style.

> We have only string as a extra fields values 

not true, you defined object there in the yaml

Reply:
added [a commit illustrating the point](https://github.com/apache/airflow/pull/15100/commits/d45157e77b941d75f2897118eaf182d76f07e491) on the test